### DISCUSSION TITLE
Using highlight just with .so files

### DISCUSSION THREAD

**Discussion by kraftwerk28** [tree-sitter/tree-sitter#889] [Q&A] [answered]
Is it possible to use `tree-sitter highlight` without first cloning language grammar repos but just using compiled .so files?

**Comment by maxbrunsfeld** [comment, marked as answer]
In addition to the parser itself, which is compiled to a `.so` file, you need a highlighting query, which is a plain text file that is parsed at runtime. See this section of the docs: https://tree-sitter.github.io/tree-sitter/syntax-highlighting#queries.

**Comment by kraftwerk28** [comment]
Thanks. I want to use treesitter for server-side highlight and interested if there's a way to avoid cloning grammar repos for every language

**Reply by maxbrunsfeld** [reply]
Yeah. In our server-side use cases, we embed the queries into our server binaries, as opposed to reading them from disk on startup. That way, our servers can be deployed as single self-contained executables.
